### ISSUE TITLE
image.hasAlpha() Method?

### ISSUE THREAD

**Issue by MCTaylor17** [oliver-moran/jimp#271]
Is it possible to add a method for determining if a PNG has a transparency colour channel?  

Creating alpha channels when none is required results in unnecessarily large images.  

Ideally, the following could be used:
```js
image.rgba( image.hasAlpha() );
```
This would either preserve the channel when used remove it when unecessary.  

Alternatively you could use it to determine file types:
```js
if ( image.hasAlpha() ) {
    file = "filename.png";
} else {
    file = "filename.jpg";
}
```

**Comment by oliver-moran** [comment]
For PNG images, you can save with an alpha channel or not using:

```js
image.rgba(false).write("image-no-alpha.png");
```

Or is it that you want to detect of there are any non-opaque pixels and decide to save as a JPEG if there are none?

**Comment by Iwasawafag** [comment]
> Or is it that you want to detect of there are any non-opaque pixels and decide to save as a JPEG if there are none?

This. 

Seems like a good idea, I'll implement it this tomorrow.

**Comment by nickgrealy** [comment]
How's this going? Looking forward to this feature!

**Comment by hipstersmoothie** [comment]
Released in [v0.4.0](https://github.com/oliver-moran/jimp/releases/tag/v0.4.0)
